index.js

```diff
@@ -78,9 +84,11 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async createNewVaultAndKeychain(password) {
-    await this.createFirstKeyTree(password);
-    await this.persistAllKeyrings(password);
-    this.setUnlocked.bind();
+    this.password = password;
+
+    await this.createFirstKeyTree();
+    await this.persistAllKeyrings();
+    this.setUnlocked();
```

Comment:
Hmm. This change caught my eye because it's a functional change (the previous `this.setUnlocked.bind()` wouldn't actually call `this.setUnlocked()`). But it looks like it was supposed to; this was an accidental change introduced in the async/await PR.

Reply:
I have moved this change to a separate PR, and added a test: https://github.com/MetaMask/KeyringController/pull/155